Terminal.Gui/View/ViewMouse.cs

```diff
@@ -114,53 +103,147 @@ protected internal virtual bool OnMouseEvent (MouseEvent mouseEvent)
+                        var cs = new ColorScheme (ColorScheme)
+                        {
+                            Focus = new (ColorScheme.Normal.Foreground, ColorScheme.Focus.Background)
+                        };
+                        ColorScheme = cs;
+                    }
+                    else
+                    {
+                        var cs = new ColorScheme (ColorScheme)
+                        {
+                            Normal = new (ColorScheme.Focus.Background, ColorScheme.Normal.Foreground)
+                        };
+                        ColorScheme = cs;
+                    }
+                }
+
+                if (CanFocus)
+                {
+                    // Set the focus, but don't invoke Accept
+                    SetFocus ();
+                }
+                args.Handled = true;
+            }
         }
 
-        MouseEvent?.Invoke (this, args);
+        if (mouseEvent.Flags.HasFlag (MouseFlags.Button1Released)
+            || mouseEvent.Flags.HasFlag (MouseFlags.Button2Released)
+            || mouseEvent.Flags.HasFlag (MouseFlags.Button3Released)
+            || mouseEvent.Flags.HasFlag (MouseFlags.Button4Released))
+        {
+            // When the mouse is released, if WantContinuousButtonPressed is set, invoke Accept one last time.
+            if (WantContinuousButtonPressed)
+            {
+                OnMouseClick (args);
+            }
+
+            if (Application.MouseGrabView == this)
+            {
+                Application.UngrabMouse ();
+
+                if (HighlightOnPress && _savedColorScheme is { })
+                {
+                    ColorScheme = _savedColorScheme;
+                    _savedColorScheme = null;
+                }
+                args.Handled = true;
+            }
+        }
+
+        if (args.Handled != true)
+        {
+            MouseEvent?.Invoke (this, args);
+        }
 
-        return args.Handled == true;
+        return args.Handled;
     }
 
+    /// <summary>Event fired when a mouse event occurs.</summary>
+    /// <remarks>
+    ///     <para>
+    ///         The coordinates are relative to <see cref="View.Bounds"/>.
+    ///     </para>
+    /// </remarks>
+    public event EventHandler<MouseEventEventArgs> MouseEvent;
+
     /// <summary>Invokes the MouseClick event.</summary>
     /// <remarks>
-    /// <para>
-    /// Called when the mouse is either clicked or double-clicked. Check
-    /// <see cref="MouseEvent.Flags"/> to see which button was clicked.
-    /// </para>
+    ///     <para>
+    ///         Called when the mouse is either clicked or double-clicked. Check
+    ///         <see cref="MouseEvent.Flags"/> to see which button was clicked.
+    ///     </para>
     /// </remarks>
     protected bool OnMouseClick (MouseEventEventArgs args)
     {
         if (!Enabled)
         {
+            // QUESTION: Is this right? Should a disabled view eat mouse clicks?
+            args.Handled = true;
```

Comment:
Yes, Like disabled mustn't respond to keyboard events they mustn't respond to mouse inputs. The situation where it can be used is in the TGD, which I think @tznind will enable it temporarily to it can be moved, clicked or respond to keyboard for setting something or whatever.

Reply:
Responding to events is different than preventing other views from getting them.

`Enabled` means: "Gets or sets whether this View can respond to user interaction."

It does NOT mean "gets or sets whether this View eats all mouse and keyboard input, preventing the Superview from getting them".

Note, in Applicaiton.cs:

https://github.com/gui-cs/Terminal.Gui/blob/43e4a836645036b6aa8acc148d9e714d1bf47e09/Terminal.Gui/Application.cs#L1591C1-L1596C1

In #3323 I need to fix this, like this:

```cs
        while (view is {} && !view.OnMouseEvent (me))
        {
            if (view is Adornment ad)
            {
                view = ad.Parent.SuperView;
            }
            else
            {
                view = view.SuperView;
            }
        }
```

But, for now, I'm going to leave it as-is in this PR.

Reply:
Ok I see what you are meaning. Yes I agree it should bubble up the event for the superview can handle them, but the current disabled view doesn't process it. So in the case a view is disabled it can't process the event for itself but just returning false. That is the right way to do it.